lua/dap/repl.lua

```diff
@@ -197,7 +214,45 @@ local function print_threads(threads)
 end
 
 
+---- Handle multiline inputs
+--
+-- Returns nil if input is not complete, otherwise returns the full user input
+local function handle_followup_input(curr_text)
+  local session = get_session()
+  if not session then
+    return curr_text
+  end
+  local continue_prompt_fn = session.adapter.followup_func_cb
+  if not continue_prompt_fn then
+    return curr_text
+  end
+  local prompt_state = vim.fn.getbufvar(repl.buf, PROMPT_STATE)
+  if prompt_state == REGULAR_PROMPT and continue_prompt_fn(curr_text) then
+    vim.fn.setbufvar(repl.buf, PROMPT_STATE, FOLLOWUP_PROMPT)
+    vim.fn.setbufvar(repl.buf, 'curr_text', curr_text .. '\n')
+    vim.fn.prompt_setprompt(repl.buf, get_prompt())
+    return nil
+  elseif prompt_state == FOLLOWUP_PROMPT then
+    local previous_text = vim.fn.getbufvar(repl.buf, 'curr_text')
+    if curr_text == '' then
+      curr_text = previous_text .. curr_text
+      vim.fn.setbufvar(repl.buf, PROMPT_STATE, REGULAR_PROMPT)
+      vim.fn.prompt_setprompt(repl.buf, get_prompt())
+      return curr_text
+    else
+      vim.fn.setbufvar(repl.buf, 'curr_text', previous_text .. curr_text .. '\n')
+      return nil
+    end
```

Comment:
Might be simpler to use a table/list as type for the variable and add the lines to the list.